addons/osio-addon/src/main/java/io/fabric8/launcher/osio/steps/booster/GolangBooster.java

```diff
@@ -0,0 +1,179 @@
+package io.fabric8.launcher.osio.steps.booster;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.lang3.math.NumberUtils;
+
+/**
+ * Customizes golan booster files.
+ *
+ */
+public class GolangBooster {
+    private static final Logger log = Logger.getLogger(GolangBooster.class.getName());
+    private static final String GO_FILE_EXTENSION = ".go";
+    private static final String ENVIRONMENT = "environment";
+    private static final String MAKEFILE = "Makefile";
+    private static final String GIT = "git";
+    private static final String URL = "url";
+    private static final String SOURCE = "source";
+    private static final String START_IMPORT = "import (";
+    private static final String END_IMPORT = ")";
+    private static final String NEW_LINE = "\n";
+    private static final String NO_FILE_EXTENSION = "";
+    private static final String SEPARATOR = "/";
+
+    private Path projectLocation;
+    private Map<String, Object> boosterData;
+    private String gitUser;
+    private Map<File, String> filesToPush = new HashMap<>();
+
+    /**
+     * C'tor will set the project location, booster data and git user.
+     *
+     * @param projectLocation
+     *            The project location where this projectile is
+     * @param boosterData
+     *            The booster data which this projectile refers to
+     * @param gitUser
+     *            The user logged into the GitService
+     */
+    public GolangBooster(Path projectLocation, Map<String, Object> boosterData, String gitUser) {
+        this.projectLocation = projectLocation;
+        this.boosterData = boosterData;
+        this.gitUser = gitUser;
+    }
+
+    /**
+     * Gets a list of files from a directory.
+     *
+     * @return a Map containing the modified file contents
+     */
+    public Map<File, String> customize() {
+        File[] directoryContents = new File(this.projectLocation.toString()).listFiles();
+        if (directoryContents != null) {
+            getBoosterFiles(directoryContents);
+        }
+
+        return this.filesToPush;
+    }
+
+    /**
+     * Recursively gets files to template.
+     *
+     * @param files
+     *            Files to template
+     */
+    private void getBoosterFiles(File[] directoryContents) {
+        for (File path : directoryContents) {
```

Comment:
You can use `java.nio.file.Files.walkFileTree` instead. See https://github.com/fabric8-launcher/launcher-backend/blob/master/core/core-impl/src/main/java/io/fabric8/launcher/core/impl/preparers/ChangeArquillianConfigurationPreparer.java#L51 for example

Reply:
Good catch, this is better on performance! Will make this change.